changelog.d/15647.bugfix

```diff
@@ -0,0 +1 @@
+Instrument `state` and `state_group` storage-related operations to better picture what's happening when tracing.
```

Comment:
Using same changelog as [#15610](https://github.com/matrix-org/synapse/pull/15610/files#diff-6e515bb2cf0373c6678f58850764d4bc4259558bf059ce6e9be6a3be2cef4945R1) so the changelog entry merges